packages/headless/src/features/search/search-actions-thunk-processor.ts

```diff
@@ -180,9 +181,10 @@ export class AsyncSearchThunkProcessor<RejectionType> {
 
     const originalQuery = this.getCurrentQuery();
     const {correctedQuery} = successResponse.queryCorrections[0];
-    const retried = await this.automaticallyRetryQueryWithCorrection(
-      correctedQuery
-    );
+
+    const retried = state.didYouMean?.automaticallyCorrectQuery
```

Comment:
This additional check here should not be needed at this point of the function: We should return early in the function when we should not auto retry.

Also, returning the same `fetched` object is... weird. 

It's kind of tricking upstream function call to think there was an automatic query correction applied (the `processor` part of this module), when in fact there were not. 

And then we log analytics as if it was a new query performed. That does not seem right.